### DISCUSSION TITLE
Why adding role in laravel-permission I got invalid table table name error?

### DISCUSSION THREAD

**Discussion by sergeynilov** [spatie/laravel-permission#2027] [Q&A] [answered]
In laravel 9 / spatie/laravel-permission 5.5 I add new role with :

    $role = Role::create([
        'name' => $request->name,
        'guard_name' => 'web',
    ]);

and I got error :
 Base table or view not found: 1146 Table 'BiCurrencies.roles' doesn't exist
 
 
 
In config/permission.php I have option :

    'table_names' => [
        'roles' => 'spt_roles',
        ...

and I Models/Role.php I set :

    class Role extends Model implements RoleContract
    {
        use HasPermissions;
        use RefreshesPermissionCache;
    
        protected $guarded = ['id'];
        protected $primaryKey = 'id';
        public $timestamps = false;
        protected $table = 'spt_roles';

        

Why error and how my config options must be applied ?

Thanks!

**Comment by erikn69** [comment, marked as answer]
You have to set the new model on config too

**Reply by sergeynilov** [reply]
Thanks!
 As I set custom models I had to set them in file config/permission.php
```
return [
    'models' => [
        'permission' => App\Models\Permission::class,
        'role' => App\Models\Role::class,
    ],

```
it works for me.
